sunpy/map/maputils.py

```diff
@@ -153,21 +164,39 @@ def contains_full_disk(smap):
     within the field of the view of the instrument (although no emission
     from the disk itself is present in the data.)
     """
-    # Calculate all the edge pixels
-    edges = map_edges(smap)
-    edge_pixels = list(chain.from_iterable([edges[0], edges[1], edges[2], edges[3]]))
-    x = [p[0] for p in edge_pixels] * u.pix
-    y = [p[1] for p in edge_pixels] * u.pix
-
-    # Calculate the edge of the world
-    edge_of_world = smap.pixel_to_world(x, y)
-
+    edge_of_world = _edge_coordinates(smap)
     # Calculate the distance of the edge of the world in solar radii
     coordinate_angles = np.sqrt(edge_of_world.Tx ** 2 + edge_of_world.Ty ** 2)
 
     # Test if all the edge pixels are more than one solar radius distant
     # and that the whole map is not all off disk.
-    return np.all(coordinate_angles > solar_angular_radius(edge_of_world)) and ~is_all_off_disk(smap)
+    return np.all(coordinate_angles > solar_angular_radius(edge_of_world)) and contains_solar_center(smap)
+
+
+def contains_solar_center(smap):
+    """
+    Returns `True` if smap contains the solar center.
+
+    This is the case if and only if the solar center is inside the edges of the map. This
+    is checked by seeing if the sign of both the x and y coordintaes of the corners are opposite
+    (ie. the (0, 0) point is contained within the map).
```

Comment:
Maybe I'm missing something, but why wouldn't you do a `.world_to_pixel()` for Helioprojective (0, 0) and see if it lands in your pixel array?  Does that approach perform worse?

Reply:
Hah, no reason that I can think of, maybe I was being a bit *too* clever here 😆. I'll try `.world_to_pixel()` on (0, 0) and compare the performance.